tests/compas/test_json.py

```diff
@@ -0,0 +1,71 @@
+import compas
+
+from compas.geometry import Point, Vector, Frame
+from compas.geometry import Box
+from compas.geometry import Transformation
+
+from compas.datastructures import Mesh
+
+
+def test_json_native():
+    before = [[], (), {}, '', 1, 1.0, True, None]
+    s = compas.json_dumps(before)
+    after = compas.json_loads(s)
+    assert after == [[], [], {}, '', 1, 1.0, True, None]
+
+
+if not compas.IPY:
+    import numpy as np
+
+    def test_json_numpy():
+        before = [np.array([1, 2, 3]), np.array([1.0, 2.0, 3.0]), np.float64(1.0), np.int32(1)]
+        s = compas.json_dumps(before)
+        after = compas.json_loads(s)
+        assert after == [[1, 2, 3], [1.0, 2.0, 3.0], 1.0, 1]
+
+
+def test_json_primitive():
+    before = Point(0, 0, 0)
+    s = compas.json_dumps(before)
+    after = compas.json_loads(s)
+    if not compas.IPY:
+        assert type(before) == type(after)
```

Comment:
I'm not a fan of skipping tests based on the environment they are run on.  With the numpy situation, it's clear that that has to happen, but here, it's not so clear.  What about `assert before.__class__ == after.__class__` or ` assert isinstance(after, before.__class__)`?